src/topp/MaRaClusterAdapter.cpp

```diff
@@ -0,0 +1,514 @@
+      }
+    }
+    return scan_identifier.removeWhitespaces();
+  }
+
+  //   replace with PercolatorAdapter function
+  Int getScanNumber_(String scan_identifier)
+  {
+    Int scan_number = 0;
+    StringList fields = ListUtils::create<String>(scan_identifier);
+    for (StringList::const_iterator it = fields.begin(); it != fields.end(); ++it)
+    {
+      // if scan number is not available, use the scan index
+      Size idx = 0;
+      if ((idx = it->find("scan=")) != string::npos)
+      {
+        scan_number = it->substr(idx + 5).toInt();
+        break;
+      }
+      else if ((idx = it->find("index=")) != string::npos)
+      {
+        scan_number = it->substr(idx + 6).toInt();
+        break;
+      }
+      else if ((idx = it->find("spectrum=")) != string::npos)
+      {
+        scan_number = it->substr(idx + 9).toInt();
+      }
+    }
+    return scan_number;
+  }
+
+  ExitCodes main_(int, const char**) override
+  {
+    //-------------------------------------------------------------
+    // parsing parameters
+    //-------------------------------------------------------------
+    const StringList in_list = getStringList_("in");
+
+    const String maracluster_executable(getStringOption_("maracluster_executable"));
+    writeDebug_(String("Path to the maracluster executable: ") + maracluster_executable, 2);
+    if (maracluster_executable.empty())  //TODO? - TOPPBase::findExecutable after registerInputFile_("maracluster_executable"... ???
+    {
+      writeLog_("No maracluster executable specified. Aborting!");
+      printUsage_();
+      return ILLEGAL_PARAMETERS;
+    }
+    else if (!File::exists(maracluster_executable))
+    {
+      writeLog_("Maracluster executable not found at location specified. Aborting!");
+      printUsage_();
+      return ILLEGAL_PARAMETERS;
+    }
+    
+    const String consensus_out(getStringOption_("consensus_out"));
+    const String out(getStringOption_("out"));
+
+    if (in_list.empty())
+    {
+      writeLog_("Fatal error: no input file given (parameter 'in')");
+      printUsage_();
+      return ILLEGAL_PARAMETERS;
+    }
+
+    if (consensus_out.empty() && out.empty())
+    {
+      writeLog_("Fatal error: no output file given (parameter 'out' or 'consensus_out')");
+      printUsage_();
+      return ILLEGAL_PARAMETERS;
+    }
+
+    //-------------------------------------------------------------
+    // read input
+    //-------------------------------------------------------------
+
+    // create temp directory to store maracluster temporary files
+    String temp_directory_body = QDir::toNativeSeparators((File::getTempDirectory() + "/" + File::getUniqueName() + "/").toQString()); // body for the tmp files
+    {
+      QDir d;
+      d.mkpath(temp_directory_body.toQString());
+    }
+    double pcut = getDoubleOption_("pcut");
+
+    String txt_designator = File::getUniqueName();
```

Comment:
QFile? Check all occurrences of File.

Reply:
Alright, working on it. Why is this necessary? The TOPPBase.cpp uses File::getUniqueName() as well..

Reply:
Hmm then an include is missing? http://cdash.openms.de/viewBuildError.php?buildid=264161